packages/sysweb3-keyring/src/eth-manager.ts

```diff
@@ -49,14 +54,22 @@ export const Web3Accounts = (): IWeb3Accounts => {
 
   const getBalance = async (address: string) => {
     try {
-      const balance = await web3Provider.getBalance(address);
+      const { network } = getDecryptedVault();
+
+      const balance = await jsonRpcRequest(network.url, 'eth_getBalance', [
+        address,
+        'latest',
+      ]);
+
+      if (!balance) return 0;
+
       const formattedBalance = ethers.utils.formatEther(balance);
 
       const roundedBalance = _.floor(parseFloat(formattedBalance), 4);
 
       return roundedBalance;
     } catch (error) {
-      return 0;
+      throw new Error(`Could not get wallet balance. Error: ${error}`);
```

Comment:
Just throw the raw error message, that way we can use pali JsonRPCError to format it and display it for the user to log in as well.
`throw error`